From 3b2b302ed75be36d604a47aafbb928d252494edb Mon Sep 17 00:00:00 2001
From: Rajkumar Ayyasamy <arajkuma@qti.qualcomm.com>
Date: Thu, 19 Mar 2026 09:06:35 +0530
Subject: [PATCH] i2c: qcom-geni: enable_irq if PM is disabled

i2c core clk and se clock is disabled at end of probe which when
PM is enabled, pm runtime takes care of enabling the clocks when
i2c bus access is detected.

If PM is disabled, clocks should be left ON. Also irq need to be
enabled.

Change-Id: I6347144c7b791260601f01152ae752a5c832e6cd
Signed-off-by: Rajkumar Ayyasamy <arajkuma@qti.qualcomm.com>
---
 drivers/i2c/busses/i2c-qcom-geni.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/i2c/busses/i2c-qcom-geni.c b/drivers/i2c/busses/i2c-qcom-geni.c
index 85123066a35d..a34a0c69753f 100644
--- a/drivers/i2c/busses/i2c-qcom-geni.c
+++ b/drivers/i2c/busses/i2c-qcom-geni.c
@@ -954,6 +954,7 @@ static int geni_i2c_probe(struct platform_device *pdev)
 		dev_dbg(dev, "i2c fifo/se-dma mode. fifo depth:%d\n", tx_depth);
 	}
 
+#ifdef CONFIG_PM
 	clk_disable_unprepare(gi2c->core_clk);
 	ret = geni_se_resources_off(&gi2c->se);
 	if (ret) {
@@ -970,6 +971,9 @@ static int geni_i2c_probe(struct platform_device *pdev)
 	pm_runtime_set_autosuspend_delay(gi2c->se.dev, I2C_AUTO_SUSPEND_DELAY);
 	pm_runtime_use_autosuspend(gi2c->se.dev);
 	pm_runtime_enable(gi2c->se.dev);
+#else
+	enable_irq(gi2c->irq);
+#endif /* CONFIG_PM */
 
 	ret = i2c_add_adapter(&gi2c->adap);
 	if (ret) {
-- 
2.34.1

